src/redux/actions.ts

```diff
@@ -0,0 +1,1307 @@
+        nbVoltageLevels: nbVoltageLevels,
+    };
+}
+
+export const SET_COMPUTING_STATUS = 'SET_COMPUTING_STATUS';
+export type SetComputingStatusAction = Readonly<
+    Action<typeof SET_COMPUTING_STATUS>
+> & {
+    computingType: ComputingType;
+    runningStatus: RunningStatus;
+};
+export function setComputingStatus(
+    computingType: ComputingType,
+    runningStatus: RunningStatus
+): SetComputingStatusAction {
+    return {
+        type: SET_COMPUTING_STATUS,
+        computingType: computingType,
+        runningStatus: runningStatus,
+    };
+}
+
+export const SET_COMPUTATION_STARTING = 'SET_COMPUTATION_STARTING';
+export type SetComputationStartingAction = Readonly<
+    Action<typeof SET_COMPUTATION_STARTING>
+> & {
+    computationStarting: boolean;
+};
+export function setComputationStarting(
+    computationStarting: boolean
+): SetComputationStartingAction {
+    return {
+        type: SET_COMPUTATION_STARTING,
+        computationStarting: computationStarting,
+    };
+}
+
+export const SET_STUDY_INDEXATION_STATUS = 'SET_STUDY_INDEXATION_STATUS';
+export type SetStudyIndexationStatusAction = Readonly<
+    Action<typeof SET_STUDY_INDEXATION_STATUS>
+> & {
+    studyIndexationStatus: StudyIndexationStatus;
+};
+export function setStudyIndexationStatus(
+    studyIndexationStatus: StudyIndexationStatus
+): SetStudyIndexationStatusAction {
+    return {
+        type: SET_STUDY_INDEXATION_STATUS,
+        studyIndexationStatus: studyIndexationStatus,
+    };
+}
+
+export const SET_OPTIONAL_SERVICES = 'SET_OPTIONAL_SERVICES';
+export type SetOptionalServicesAction = Readonly<
+    Action<typeof SET_OPTIONAL_SERVICES>
+> & {
+    optionalServices: IOptionalService[];
+};
+export function setOptionalServices(
+    optionalServices: IOptionalService[]
+): SetOptionalServicesAction {
+    return {
+        type: SET_OPTIONAL_SERVICES,
+        optionalServices: optionalServices,
+    };
+}
+
+export const SET_ONE_BUS_SHORTCIRCUIT_ANALYSIS_DIAGRAM =
+    'SET_ONE_BUS_SHORTCIRCUIT_ANALYSIS_DIAGRAM';
+export type SetOneBusShortcircuitAnalysisDiagramAction = Readonly<
+    Action<typeof SET_ONE_BUS_SHORTCIRCUIT_ANALYSIS_DIAGRAM>
+> &
+    (OneBusShortCircuitAnalysisDiagram | { diagramId: null });
+
+export function setOneBusShortcircuitAnalysisDiagram(
+    diagramId: null
+): SetOneBusShortcircuitAnalysisDiagramAction;
+export function setOneBusShortcircuitAnalysisDiagram(
+    diagramId: OneBusShortCircuitAnalysisDiagram['diagramId'],
+    nodeId: OneBusShortCircuitAnalysisDiagram['nodeId']
+): SetOneBusShortcircuitAnalysisDiagramAction;
+export function setOneBusShortcircuitAnalysisDiagram(
+    diagramId: OneBusShortCircuitAnalysisDiagram['diagramId'] | null,
+    nodeId?: OneBusShortCircuitAnalysisDiagram['nodeId']
+): SetOneBusShortcircuitAnalysisDiagramAction {
+    return {
+        type: SET_ONE_BUS_SHORTCIRCUIT_ANALYSIS_DIAGRAM,
+        diagramId: diagramId,
+        // @ts-expect-error: function overload protect call
```

Comment:
Well, it's because I haven't done in full.

Because of the function overload (the 2 declarations before the implementation), we known that if we call _setFullScreenDiagram_ with null that tsc will not let us pass the 2nd parameter ; and if we pass a string the second parameter is mandatory. So I haven't judge necessary to do the full version without warning that is
```typescript
export function setFullScreenDiagram(
    diagramIdParam: string | null,
    svgTypeParam?: DiagramType
): SetFullscreenDiagramAction {
    if (diagramIdParam === null) {
        return {
            type: SET_FULLSCREEN_DIAGRAM,
            diagramId: diagramIdParam,
        };
    } else {
        return {
            type: SET_FULLSCREEN_DIAGRAM,
            diagramId: diagramIdParam,
            svgType: svgTypeParam!,
        };
    }
}
```
which still ask to say in the else "trust me it's not undefined in this case"...